frontend/pages/bookmark.tsx

```diff
@@ -0,0 +1,353 @@
+  }
+`;
+
+const SvgUnbookmarkedStyled = styled(SvgUnbookmarked)`
+  cursor: pointer;
+
+  @media ${device.laptop} {
+    left: 85%;
+    position: absolute;
+    top: 2%;
+    transform: scale(2);
+  }
+`;
+
+const ImageContainer = styled.div`
+  padding-right: 10px;
+`;
+
+const Title = styled.h1`
+  display: inline;
+  font-size: 22px;
+  margin: 0;
+
+  @media ${device.laptop} {
+    font-size: 48px;
+    font-weight: 900;
+    line-height: 56px;
+    letter-spacing: -0.005em;
+  }
+`;
+
+// TODO: ask for all icons
+const MethodIconsMap: Record<string, ReactElement> = {
+  Implant: <SvgImplantBookmark />,
+  Patch: <SvgPatchBookmark />,
+  Condom: <SvgCondomBookmark />,
+  'Copper IUD': <SvgImplantBookmark />,
+  'Hormonal IUD': <SvgImplantBookmark />,
+  Sterilization: <SvgImplantBookmark />,
+  Shot: <SvgPatchBookmark />,
+  Ring: <SvgPatchBookmark />,
+  Spermicide: <SvgPatchBookmark />,
+  Diaphragm: <SvgPatchBookmark />,
+  Pill: <SvgPatchBookmark />,
+  'Cervical Cap': <SvgPatchBookmark />,
+};
+
+interface MethodProps {
+  icon: ReactElement;
+  name: string;
+  effectiveRate: number;
+  usePatternHighBound: number;
+  usePatternUnits: TimeUnits;
+  costMin: number;
+  costMax: number;
+  whoAdministers: string;
+}
+
+const Method = ({
+  icon,
+  name,
+  effectiveRate,
+  usePatternHighBound,
+  usePatternUnits,
+  costMin,
+  costMax,
+  whoAdministers,
+}: MethodProps): ReactElement => {
+  const [bookmarked, setBookmarked] = useState<Boolean>(true);
+  const bookmarkClicked = (name: string) => {
+    setBookmarked(!bookmarked);
+    if (!bookmarked) {
+      API.user.postBookmark(name);
+    } else {
+      console.log(name);
+      API.user.deleteBookmark(name);
+    }
+  };
+  const BookmarkIcon = bookmarked ? SvgBookmarkedStyled : SvgUnbookmarkedStyled;
+  return (
+    <>
+      <MethodCard>
+        <ImageContainer>{icon}</ImageContainer>
+
+        <MethodInfoColumn>
+          <MethodName>{name}</MethodName>
+
+          <MethodInfo>{effectiveRate}% Effective</MethodInfo>
+          <MethodInfo>
+            Lasts up to {usePatternHighBound} {usePatternUnits}
+          </MethodInfo>
+          <MethodInfo>
+            Can cost ${costMin} - ${costMax}
+          </MethodInfo>
+          <MethodInfo>{whoAdministers}</MethodInfo>
+        </MethodInfoColumn>
+        <ColumnExtra>
+          <BookmarkIcon onClick={() => bookmarkClicked(name)} />
+
+          <LearnMoreButton> LEARN MORE {'>'} </LearnMoreButton>
+        </ColumnExtra>
+      </MethodCard>
+    </>
+  );
+};
+type CompareProps = {
+  contraceptives: Contraceptive[];
+};
+const Bookmark = (compareProps: CompareProps): ReactElement => {
+  //const [contraceptives, setContraceptives] = useState<Contraceptive[]>([]);
+  const { contraceptives } = compareProps;
+  return (
+    <>
+      <Container>
+        <MenuBar />
+        <Header>
+          <Title>Bookmarks</Title>
+        </Header>
+        {console.log(contraceptives[0] + 'val')}
```

Comment:
Not totally sure what this log statement does

Reply:
it can be removed. For testing mostly.